x-pack/plugins/apm/public/components/shared/charts/latency_chart/index.tsx

```diff
@@ -89,6 +109,51 @@ export function LatencyChart({ height, kuery }: Props) {
   const latencyMaxY = getMaxY(timeseries);
   const latencyFormatter = getDurationFormatter(latencyMaxY);
 
+  const { setScreenContext } =
+    useApmPluginContext().observabilityAIAssistant.service;
+
+  useEffect(() => {
+    const clauses = [
+      `${PROCESSOR_EVENT} == "transaction"`,
+      getEsqlDateRangeFilter(start, end),
+      serviceName ? `${SERVICE_NAME} == ${string`${serviceName}`}` : '',
+      transactionName
+        ? `${TRANSACTION_NAME} == ${string`${transactionName}`}`
+        : '',
+      transactionType
+        ? `${TRANSACTION_TYPE} == ${string`${transactionType}`}`
+        : '',
+      environment ? getEsqlEnvironmentFilter(environment) : '',
+    ].filter(Boolean);
+
+    return setScreenContext({
+      screenDescription: `There is a latency chart displayed. The ES|QL equivalent for this is:
+    
+      \`\`\`esql
+      FROM traces-apm*
+        | WHERE ${clauses.join(' AND ')}
+        ${
+          bucketSizeInSeconds !== undefined
+            ? `| EVAL date_bucket = DATE_TRUNC(${bucketSizeInSeconds} seconds, @timestamp)`
+            : ''
+        }
+        | STATS avg_duration = AVG(transaction.duration.us)${
+          bucketSizeInSeconds !== undefined ? ` BY date_bucket` : ''
+        }
+      \`\`\`
+    `,
+    });
+  }, [
+    serviceName,
+    transactionName,
+    transactionType,
+    environment,
+    setScreenContext,
+    bucketSizeInSeconds,
+    start,
+    end,
+  ]);
+
```

Comment:
I wonder if we should start creating custom hooks to clean up the components. Something like `useLatencyChartScreenContext` and move all this code there.

Reply:
I think that's good advice, will refactor both usages of `setScreenContext`.